build/toolchain/gcc_toolchain.gni

```diff
@@ -48,11 +46,7 @@ template("gcc_toolchain") {
       cxx = invoker.cxx
     }
 
-    if (defined(invoker.link_generate_map_file)) {
```

Comment:
It seems like the goal of the patch can be accomplished with just the one line change defaulting chip_generate_link_map_file to false.

What was the reason to change the way that argument works?

Reply:
> It seems like the goal of the patch can be accomplished with just the one line change defaulting chip_generate_link_map_file to false.

The `chip_generate_link_map_file` was introduced in `d15daf8a02b97` to allow disabling map files for Python builds. Also, the `link_generate_map_file` is here to allow disabling map files for Android. So, there are two mechanisms which basically do the same but in a different way. This PR unifies this by removing one of these (the less generic one, which is `link_generate_map_file` defined by the toolchain). Also, it changes the default behavior since map files are rather big and if not required they should be disabled.

Reply:
Assuming this explanation resolves the comment (unifying mechanisms and also fixing up builder logic results in a larger PR).